### PR TITLE
[package-json-validator] add types

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

https://github.com/TechNickAI/package.json-validator/tree/f1fe16c84ad9a32d4d98a6f9864a703fb3693f04 -> https://github.com/TechNickAI/package.json-validator/blob/master/PJV.js

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/package-json-validator/package-json-validator-tests.ts
1. types/package-json-validator/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7674dbcc3fbe48e39fbcf6c79ab1774ba6a3cae6
</p>

File: types/package-json-validator/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "private": true,
+    "name": "@types/package-json-validator",
+    "version": "0.6.9999",
+    "projects": [
+        "http://package-json-validator.com"
```
Comment (jakebailey) [inline types/package-json-validator/package.json:6]: This is a dead page for me; is there a github link?

Comment (JoshuaKGoldberg) [reply types/package-json-validator/package.json]: Lovely. Updated to https://github.com/TechNickAI/package.json-validator, filed https://github.com/TechNickAI/package.json-validator/issues/76, and sent https://github.com/TechNickAI/package.json-validator/pull/77.

**Review by jakebailey — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
Ready to merge

**Pull request merged**